sql/src/test/java/org/apache/druid/sql/calcite/BaseCalciteQueryTest.java

```diff
@@ -1005,28 +1030,31 @@ protected static boolean isRewriteJoinToFilter(final Map<String, Object> queryCo
     );
   }
 
+
   /**
    * Override not just the outer query context, but also the contexts of all subqueries.
    */
-  public static <T> Query<T> recursivelyOverrideContext(final Query<T> query, final Map<String, Object> context)
+  public static <T> Query<T> recursivelyClearContext(final Query<T> query)
   {
-    return query.withDataSource(recursivelyOverrideContext(query.getDataSource(), context))
-                .withOverriddenContext(context);
+    Query<T> newQuery = query.withDataSource(recursivelyClearContext(query.getDataSource()));
+    newQuery.context().clear();
```

Comment:
We need to pass in an empty context to a copy of the native query if we want to clear the context.